draft-brunstrom-taps-impl.md

```diff
@@ -330,6 +343,35 @@ Protocol options are checked next in order. Whether or not a set of protocol, or
 
 Branching for derived endpoints is the final step, and may have multiple layers of derivation or resolution, such as DNS service resolution and DNS hostname resolution.
 
+
+## Ranking Branches
+
+Implementations SHOULD rank the branches of the tree of connection options in order to prioritize them for candidate racing. Leaf nodes on branches with higher rankings represent connection attempts that will be raced first. Implementations SHOULD order the branches to reflect the preferences expressed by the application for its new connection, including Protocol and Path Selection Properties as well as Application Intents, which are specified in {{draft-trammell-taps-interface}}. In addition to the properties provided by the application, an implementation MAY include additional criteria such as cached performance estimates, see {{performance-caches}}, or system policy, see {{role-of-system-policy}}, in the ranking.
+
+The Transport Parameters specified in {{draft-trammell-taps-interface}} may be used to rank branches in the following ways:
+
+* Interface Type:
+If the application specifies an interface type to be required, an implementation MUST purge all branches that correspond to paths over interfaces that do not match this type. Similarly, it MUST purge paths that correspond to interface types that the application prohibits. If the application specifies an interface type to be preferred or avoided, implementations SHOULD rank paths accordingly.
+
+* Capacity Profile:
+An implementation may use the Capacity Profile to prefer paths optimized for the application's expected traffic pattern according to cached performance estimates, see {{performance-caches}}:
+   * Interactive/Low Latency:
+     Prefer paths with the lowest expected Round Trip Time
+   * Constant Rate:
+     Prefer paths that can satisfy the requested Stream Send or Stream Receive Bitrate, based on observed maximum throughput
+   * Scavenger/Bulk:
+     Prefer paths with the highest expected available bandwidth, based on observed maximum throughput
+
+* Size to be Sent or Received:
```

Comment:
Please remove items that are still just in the appendix and not agreed upon for now. (that's Size, Bitrate, and Cost)

Reply:
I guess we can move them to an appendix for now - completely removing them makes this pull-request pointless.